.github/workflows/coverage-reports.yml

```diff
@@ -0,0 +1,118 @@
+# ------------------------------------------------------------
+# Copyright 2021 The Dapr Authors
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------
+
+name: Publishing test coverage reports of components
+
+on:
+  repository_dispatch:
+    types: [coverage-report]
+  workflow_dispatch:
+  schedule:
+    - cron: '0 0 * * 0'
+
+jobs:
+  coverage:
+      name: Download and merge gocov reports
+      runs-on: ubuntu-latest
+      env:
+        GOVER: "1.20"
+      steps:
+        - name: Set up Go ${{ env.GOVER }}
+          if: ${{ steps.skip_check.outputs.should_skip != 'true' }}
+          uses: actions/setup-go@v3
+          with:
+            go-version: ${{ env.GOVER }}
+
+        - name: Check out code
+          uses: actions/checkout@v3
+          with:
+            repository: ${{ env.CHECKOUT_REPO }}
+            ref: ${{ env.CHECKOUT_REF }}
+
+        - name: Setup go dependencies
+          run: |
+            go install github.com/axw/gocov/gocov@v1.1.0
+            go install gotest.tools/gotestsum@latest
```

Comment:
Should we pin a version for gotestsum and gocovmerge? Also for security reasons

Reply:
Done